pkg/server/plugin/notifier/k8sbundle/k8sbundle.go

```diff
@@ -155,28 +160,84 @@ func (p *Plugin) getConfig() (*pluginConfig, error) {
 	return p.config, nil
 }
 
-func (p *Plugin) setConfig(config *pluginConfig) {
+func (p *Plugin) setConfig(config *pluginConfig) error {
 	p.mu.Lock()
 	defer p.mu.Unlock()
 	p.config = config
+
+	// Start watcher to set CA Bundle in objects created after server has started
+	if p.cancelWatcher != nil {
+		p.cancelWatcher()
+	}
+	if config.Label != "" {
+		watcher, err := newBundleWatcher(p, config)
+		if err != nil {
+			return err
+		}
+		var wg sync.WaitGroup
+		ctx, cancel := context.WithCancel(context.Background())
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			if err := watcher.Watch(ctx); err != nil && !errors.Is(err, context.Canceled) {
+				p.log.Error("Unable to watch: %v", err)
+			}
+		}()
+		p.cancelWatcher = func() {
+			cancel()
+			wg.Wait()
+		}
+	}
+
+	return nil
 }
 
-func (p *Plugin) updateBundleConfigMap(ctx context.Context, c *pluginConfig) (err error) {
-	client, err := p.hooks.newKubeClient(c.KubeConfigFilePath)
+// updateBundles iterates through all the objects that need an updated CA bundle
+func (p *Plugin) updateBundles(ctx context.Context, c *pluginConfig) (err error) {
+	clients, err := p.hooks.newKubeClient(c)
 	if err != nil {
 		return err
 	}
 
-	for {
-		// Get the config map so we can use the version to resolve conflicts racing
+	var updateErrs string
+	for _, client := range clients {
+		list, err := client.GetList(ctx, c)
+		if err != nil {
+			return k8sErr.New("unable to get list: %v", err)
```

Comment:
should this error also just be aggregated?

Reply:
all these errors are now aggregated.